### PR TITLE
Dev #76 - Refactor HackShack landing site architecture

### PR DESCRIPTION
**Pull request by halocline**
## Scope of PR
Refactor site architecture to reflect the following:

+-- hpe-hack-shack-lv-19
..... +-- landing-page (Client 1)
.......... +-- node_modules
.......... +-- src
.......... +-- package.json
.......... +-- ....
..... +-- leader-board (Client 2)
.......... +-- .next
.......... +-- node_modules
.......... +-- package.json
.......... +-- ...
..... +-- data (Shared data across clients)
..... +-- node_modules
..... +-- package.json

## PR Changes Overview

1. Created /data directory to share data across apps.
2. Moved landing page app from root to /landing-page
3. Copied '/frontend' app from 'https://github.com/HewlettPackard/hpe-hack-shack-attack.git' and placed in '/leader-board' directory.
  * NOTE: There is still an outstanding PR for the Leaderboard app. Once this is reviewed/accepted we will need to replace /leader-board in this repo. Then remove /frontend from the hpe-hack-shack-attack repo
4. Added 'dev' script to package.json to run client app development builds in parallel.
5. Implemented react-app-rewired to /landing-page app to override create-react-app ModuleScopePlugin.
6. Moved data files from each app to project root /data. 
7. Updated README.md Getting Started and added project ToDos to capture known work to be completed.

### CODE REVIEW

**Comment by halocline** [comment]
@IanKBovard -- Yes, we can do this under a single client.

**Comment by IanKBovard** [comment]
I can help with that if you'd like. should i open a new issue, or should we handle it on this branch?

**Comment by halocline** [comment]
I guess it depends on when you are going to start working on it vs. when my
refactor PR is merged. If you are getting started before the merge, we
should add the carousel and supporting views for the leaderboard on this
branch. Agree?

**Comment by IanKBovard** [comment]
Makes sense. Since, i'll working on finishing up the game today, and some of tomorrow, i can wait for this PR with these new changes to be merged in first.

File: data/events.json [hunk 1]
```diff
@@ -0,0 +1,89 @@
+[
```
Comment (karatechops) [inline data/events.json]: Is this file still needed? I would think the `hpe-discover-events.json` would be the single source of truth.

File: landing-page/src/containers/App.js [hunk 2]
```diff
@@ -6,9 +6,13 @@ import {
...
 import Header from '../components/Header';
 import TabLayout from '../components/CardLayout/index';
-import Day1 from '../data/day1.json';
-import Day2 from '../data/day2.json';
-import Day3 from '../data/day3.json';
+// import Day1 from '../../../data/day1.json';
+// import Day2 from '../../../data/day2.json';
+// import Day3 from '../../../data/day3.json';
+import eventSchedule from '../../../data/hpe-discover-events.json';
+const Day1 = eventSchedule.filter( session => session.date === 'Tuesday, June 18, 2019');
```
Comment (karatechops) [inline landing-page/src/containers/App.js]: Linter should be picking up the space after the open parens. If not, please add a rule to the linter.

File: leader-board/config.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export const Config = {
```
Comment (karatechops) [inline leader-board/config.js]: Needs copyright

File: leader-board/next.config.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+/* eslint-disable */
```
Comment (karatechops) [inline leader-board/next.config.js]: Needs copyright

File: leader-board/pages/leaderboard.js [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+/* eslint-disable no-underscore-dangle */
+
+import React, { Component } from 'react';
+import {
+  Box, Heading, Text, Grommet,
+} from 'grommet';
+import { Config } from '../config';
+import PostData from '../../data/events.json';
+
+const theme = {
```
Comment (karatechops) [inline leader-board/pages/leaderboard.js]: theme should be refactored to a separate `theme.js` file.

File: leader-board/pages/leaderboard.js [hunk 6]
```diff
@@ -0,0 +1,179 @@
+/* (C) Copyright 2019 Hewlett Packard Enterprise Development LP. */
+/* eslint-disable no-underscore-dangle */
```
Comment (karatechops) [inline leader-board/pages/leaderboard.js]: Why is the required? As best practice, try to only disable singe lines.

File: leader-board/pages/leaderboard.js [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+              upcoming sessions{' '}
+            </Heading>
+            <Box
+              margin="small"
+              gap="medium"
+              width="large"
+              align="center"
+              alignSelf="center"
+            >
+
```
Comment (karatechops) [inline leader-board/pages/leaderboard.js]: Linter should be picking up this empty line. If not, please add a rule.

File: leader-board/pages/leaderboard.js [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+    };
+  }
+
+  componentDidMount() {
+    this.interval = setInterval(() => {
+      const newDate = new Date();
+      this.setState({ currentDay: newDate.getDate().toString() });
+    }, 1000);
+
+    this._isMounted = true;
```
Comment (karatechops) [inline leader-board/pages/leaderboard.js]: Can I get some rationale behind the `_isMounted` boolean? It appears extraneous.

File: data/events.json (hunk 1 above)
Comment (halocline) [reply data/events.json]: @karatechops Agree. hpe-discover-events.json will be the single source. Other .json will be removed.

File: data/hpe-discover-events.json [hunk 9]
```diff
@@ -0,0 +1,408 @@
+[
+  {
+    "id": "18626_8949",
+    "session_id": "HSW8626",
+    "title": "HPE Design Workshop: Design thinking for disruptive ideas, Empathy and Brainstorming",
+    "track": "Hybrid Cloud ; Software-defined Infrastructure",
+    "date": "Tuesday, June 18, 2019",
```
[the remark this answers is not in the document]
Comment (halocline) [reply data/hpe-discover-events.json]: @karatechops Agree here as well. Grabbed this data from https://content.attend.hpe.com/go/agendabuilder.sessions/?l=39&locale=en_US just to get a sense of what was available and consumed what was easy to access. Will update with Date object in future PR.

File: leader-board/pages/leaderboard.js (hunk 5 above)
Comment (IanKBovard) [reply leader-board/pages/leaderboard.js]: I have this fixed in a branch that i'm working on.

File: leader-board/pages/leaderboard.js (hunk 8 above)
Comment (IanKBovard) [reply leader-board/pages/leaderboard.js]: Not sure of its purpose. The leaderboard has been redesigned in my current branch, and is not using it.

File: leader-board/pages/leaderboard.js (hunk 7 above)
Comment (IanKBovard) [reply leader-board/pages/leaderboard.js]: The linter is picking this up for me. Will fix this on my branch.

File: leader-board/pages/leaderboard.js (hunk 6 above)
Comment (IanKBovard) [reply leader-board/pages/leaderboard.js]: removed

File: leader-board/config.js (hunk 3 above)
Comment (IanKBovard) [reply leader-board/config.js]: added

File: leader-board/next.config.js (hunk 4 above)
Comment (IanKBovard) [reply leader-board/next.config.js]: added on my branch

File: hpe-discover-events.json [hunk 10]
```diff
@@ -0,0 +1 @@
+undefined
```
[the remark this answers is not in the document]
Comment (halocline) [reply hpe-discover-events.json]: Removed

File: data/events.json (hunk 1 above)
Comment (halocline) [reply data/events.json]: Removed

File: landing-page/src/containers/App.js (hunk 2 above)
Comment (halocline) [reply landing-page/src/containers/App.js]: Rule added.

File: data/hpe-discover-events.json (hunk 9 above)
[the remark this answers is not in the document]
Comment (halocline) [reply data/hpe-discover-events.json]: Added issue to capture this request: https://github.com/HewlettPackard/hpe-hack-shack-lv-19/issues/10
